apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java

```diff
@@ -884,18 +952,99 @@ private void addC4DicIdentifier(Coverage coverage, Beneficiary beneficiary) {
-            .setValue((String.valueOf(beneficiary.getBeneficiaryId())))
-            .setSystem(TransformerConstants.CODING_BBAPI_BENE_ID)
-            .setAssigner(new Reference(organization));
-    coverage.addIdentifier(identifier);
+    if (beneficiary.getMedicareBeneficiaryId().isPresent()) {
+      Identifier identifier =
+          new Identifier()
+              .setType(
+                  TransformerUtilsV2.createCodeableConcept(
+                      TransformerConstants.CODING_SYSTEM_HL7_IDENTIFIER_TYPE,
+                      null,
+                      TransformerConstants.PATIENT_MB_ID_DISPLAY,
+                      "MB"))
+              .setValue(beneficiary.getMedicareBeneficiaryId().get())
+              .setSystem(TransformerConstants.CODING_BBAPI_MEDICARE_BENEFICIARY_ID_UNHASHED)
+              .setAssigner(new Reference(organization));
+      coverage.addIdentifier(identifier);
+    }
+  }
+
+  /**
+   * Denotes which parts the beneficiary is subscribed to on the provided {@link Coverage} resource.
+   *
+   * @param beneficiary the value for {@link Beneficiary}
+   * @param coverage The {@link Coverage} to Coverage details
+   */
+  private void addSubscribedParts(Beneficiary beneficiary, Coverage coverage) {
+    if (beneficiary.getPartACoverageStartDate().isPresent()) {
+      createCoverageClass(
+          coverage,
+          CoverageClass.PLAN,
+          TransformerConstants.COVERAGE_PLAN_PART_A,
+          Optional.empty());
+    }
+    if (beneficiary.getPartBCoverageStartDate().isPresent()) {
+      createCoverageClass(
+          coverage,
+          CoverageClass.PLAN,
+          TransformerConstants.COVERAGE_PLAN_PART_B,
+          Optional.empty());
+    }
+    Optional<Instant> lastUpdatedOpt = beneficiary.getLastUpdated();
+    if (lastUpdatedOpt.isPresent()) {
+      Date lastUpdated = Date.from(lastUpdatedOpt.get());
+      if (beneficiaryHasPartC(beneficiary, lastUpdated.getMonth())) {
+        createCoverageClass(
+            coverage,
+            CoverageClass.PLAN,
+            TransformerConstants.COVERAGE_PLAN_PART_C,
+            Optional.empty());
+      }
+    }
+    if (beneficiary.getPartDCoverageStartDate().isPresent()) {
+      createCoverageClass(
+          coverage,
+          CoverageClass.PLAN,
+          TransformerConstants.COVERAGE_PLAN_PART_D,
+          Optional.empty());
+    }
+  }
+
+  /**
+   * Determines if the beneficiary is subscribed to Part C. Due to a lag with ingestion, we check if
+   * last_updated > current month, we check if the previous month is present.
+   *
+   * @param beneficiary the value for {@link Beneficiary}
+   * @param month The month that the beneficiary record was last updated within the database.
+   * @return true/false if beneficiary has Part C
+   */
+  private boolean beneficiaryHasPartC(Beneficiary beneficiary, int month) {
+    return switch (month) {
+      case 1 -> beneficiary.getPartCContractNumberJanId().isPresent()
```

Comment:
We may want to check prod and see if this is for sure the case, but since we're already checking against the last updated month here, I think we can omit the check for the previous month.

My initial idea was to check the current month and the last updated month, but I think just checking the last updated month should be sufficient unless there's a case where we get a load on the first week of the month that doesn't provide the contract IDs for some reason.

Reply:
@aschey-forpeople Hmm that's what i was thinking about as well but wasn't sure. If you/team think it's safe enough then I can remove the check on the previous month

Reply:
I think the way it's written here would cause it to say they still have coverage for the current month if we've already ingested a load that says they lost coverage, so I think this way is more likely to cause incorrect values.